exec.go

```diff
@@ -0,0 +1,81 @@
+package gocql
+
+import (
+	"fmt"
+)
+
+// SingleHostQueryExecutor allows to quickly execute diagnostic queries while
+// connected to only a single node.
+type SingleHostQueryExecutor struct {
+	session *Session
+	control *controlConn
+}
+
+// Query executes the specified CQL statement.
+// Caller is responsible for closing the result iterator after use.
+func (e SingleHostQueryExecutor) Query(stmt string, values ...interface{}) *Iter {
+	return e.control.query(stmt, values...)
+}
+
+func (e SingleHostQueryExecutor) Close() {
+	if e.control != nil {
+		e.control.close()
+	}
+	if e.session != nil {
+		e.session.Close()
+	}
+}
+
+// MakeSingleHostQueryExecutor creates a SingleHostQueryExecutor by connecting
+// to one of the hosts specified in the ClusterConfig.
+// If ProtoVersion is not specified version 4 is used.
+// Caller is responsible for closing the executor after use.
+func MakeSingleHostQueryExecutor(cfg *ClusterConfig) (e SingleHostQueryExecutor, err error) {
```

Comment:
Perhaps we should call this function `NewSingleHostQueryExecutor` instead of `MakeSingleHostQueryExecutor`? That way it would be more consistent with `NewSession` and other places that usually use `New` prefix.

Reply:
ok